### PR TITLE
feature[TW26355]: Add rest call to convert rels per type

### PR DESCRIPTION
**Pull request by audrey-denk**
Change-Id: I8e245150251d28f07f52ae04933e3d2508c1c13a

### CODE REVIEW

File: plugins/org.eclipse.osee.framework.core/src/org/eclipse/osee/framework/core/data/OrcsTokenServiceImpl.java [hunk 1]
```diff
@@ -46,13 +55,25 @@ public final class OrcsTokenServiceImpl implements OrcsTokenService {
...
 
+   private final List<String> convertedRelations = new ArrayList<>();
    /**
     * Register core types first to prevent their ids from being registered by mistaken or malicious code
     */
    public OrcsTokenServiceImpl() {
       artifactTypes.put(ArtifactTypeToken.SENTINEL.getId(), ArtifactTypeToken.SENTINEL);
       attributeTypes.put(AttributeTypeToken.SENTINEL.getId(), AttributeTypeToken.SENTINEL);
       relationTypes.put(RelationTypeToken.SENTINEL.getId(), RelationTypeToken.SENTINEL);
+      File relFile = OseeInf.getResourceAsFile("relations/relations.json", OrcsTypeTokens.class);
```
Comment (lucianovaglienti) [inline plugins/org.eclipse.osee.framework.core/src/org/eclipse/osee/framework/core/data/OrcsTokenServiceImpl.java:66]: Probably want a test around this just so we can make sure that it's loading on all platforms...

Comment (lucianovaglienti) [inline plugins/org.eclipse.osee.framework.core/src/org/eclipse/osee/framework/core/data/OrcsTokenServiceImpl.java:66]: I thought we said we wanted one per - relation type java file...are we going to do that later?

File: plugins/org.eclipse.osee.orcs.rest/src/org/eclipse/osee/orcs/rest/internal/RelationEndpointImpl.java [hunk 2]
```diff
@@ -91,6 +93,31 @@ public List<RelationTypeToken> convertRelations(ArtifactId artToken, RelationTyp
       return null;
    }
    
+   @Override
+   public List<RelationTypeToken> convertAllRelations(RelationTypeToken oldRelationType,
+      RelationTypeToken newRelationType) {
+      orcsApi.tokenService().registerRelationType(ShadowCoreRelationTypes.DefaultHierarchicalRel);
```
Comment (lucianovaglienti) [inline plugins/org.eclipse.osee.orcs.rest/src/org/eclipse/osee/orcs/rest/internal/RelationEndpointImpl.java:99]: Think you want to un-hardcode this

File: plugins/org.eclipse.osee.orcs.rest/src/org/eclipse/osee/orcs/rest/internal/RelationEndpointImpl.java [hunk 3]
```diff
@@ -91,6 +93,31 @@ public List<RelationTypeToken> convertRelations(ArtifactId artToken, RelationTyp
...
+      RelationTypeSide sideB = new RelationTypeSide(oldRelationType, RelationSide.SIDE_B);
+      TransactionBuilder tx = orcsApi.getTransactionFactory().createTransaction(branch,
+         String.format("Converting relations for Old Relation = %s New Relation = %s",
+            oldRelationType.getName(), oldRelationType.getName()));
+      
+      for (ArtifactReadable art : orcsApi.getQueryFactory().fromBranch(branch).andRelationExists(oldRelationType).asArtifacts()) {
+         
+            if (art.getRelatedCount(sideB) > 0) {
+            
+            for (ArtifactReadable artB : art.getRelated(sideB).getList()) {
```
Comment (lucianovaglienti) [inline plugins/org.eclipse.osee.orcs.rest/src/org/eclipse/osee/orcs/rest/internal/RelationEndpointImpl.java:109]: I think we need logic for preserving the order here...

File: plugins/org.eclipse.osee.orcs.rest/src/org/eclipse/osee/orcs/rest/internal/RelationEndpointImpl.java [hunk 4]
```diff
@@ -91,6 +93,31 @@ public List<RelationTypeToken> convertRelations(ArtifactId artToken, RelationTyp
...
    }
    
+   @Override
+   public List<RelationTypeToken> convertAllRelations(RelationTypeToken oldRelationType,
+      RelationTypeToken newRelationType) {
+      orcsApi.tokenService().registerRelationType(ShadowCoreRelationTypes.DefaultHierarchicalRel);
+      RelationTypeSide sideB = new RelationTypeSide(oldRelationType, RelationSide.SIDE_B);
+      TransactionBuilder tx = orcsApi.getTransactionFactory().createTransaction(branch,
+         String.format("Converting relations for Old Relation = %s New Relation = %s",
+            oldRelationType.getName(), oldRelationType.getName()));
```
Comment (lucianovaglienti) [inline plugins/org.eclipse.osee.orcs.rest/src/org/eclipse/osee/orcs/rest/internal/RelationEndpointImpl.java:103]: newRelationType

File: plugins/org.eclipse.osee.framework.core/src/org/eclipse/osee/framework/core/data/OrcsTokenServiceImpl.java (hunk 1 above)
Comment (audrey-denk) [reply plugins/org.eclipse.osee.framework.core/src/org/eclipse/osee/framework/core/data/OrcsTokenServiceImpl.java:66]: yes!  I need to ad the path separator business

File: plugins/org.eclipse.osee.orcs.rest/src/org/eclipse/osee/orcs/rest/internal/RelationEndpointImpl.java (hunk 2 above)
Comment (audrey-denk) [reply plugins/org.eclipse.osee.orcs.rest/src/org/eclipse/osee/orcs/rest/internal/RelationEndpointImpl.java:99]: got it

File: plugins/org.eclipse.osee.orcs.rest/src/org/eclipse/osee/orcs/rest/internal/RelationEndpointImpl.java (hunk 3 above)
Comment (audrey-denk) [reply plugins/org.eclipse.osee.orcs.rest/src/org/eclipse/osee/orcs/rest/internal/RelationEndpointImpl.java:109]: I think this does return in order, but I'll validate that.

File: plugins/org.eclipse.osee.orcs.rest/src/org/eclipse/osee/orcs/rest/internal/RelationEndpointImpl.java (hunk 4 above)
Comment (audrey-denk) [reply plugins/org.eclipse.osee.orcs.rest/src/org/eclipse/osee/orcs/rest/internal/RelationEndpointImpl.java:103]: got it

File: plugins/org.eclipse.osee.orcs.rest/src/org/eclipse/osee/orcs/rest/internal/RelationEndpointImpl.java (hunk 3 above)
Comment (lucianovaglienti) [reply plugins/org.eclipse.osee.orcs.rest/src/org/eclipse/osee/orcs/rest/internal/RelationEndpointImpl.java:109]: no...what I'm saying is that the user might have defined a custom order(i.e. on the attribute), and we need to handle that here and put the order in the relation.

File: plugins/org.eclipse.osee.framework.core/src/org/eclipse/osee/framework/core/data/OrcsTokenServiceImpl.java (hunk 1 above)
Comment (audrey-denk) [reply plugins/org.eclipse.osee.framework.core/src/org/eclipse/osee/framework/core/data/OrcsTokenServiceImpl.java:66]: I don't remember that... however, at the very least we will need one for the open source relations, lba relations and f18 relations...

**Review by jadenpuckett — APPROVED** [review]

File: plugins/org.eclipse.osee.orcs.rest/src/org/eclipse/osee/orcs/rest/internal/RelationEndpointImpl.java (hunk 3 above)
Comment (audrey-denk) [reply plugins/org.eclipse.osee.orcs.rest/src/org/eclipse/osee/orcs/rest/internal/RelationEndpointImpl.java:109]: ok, had to switch to using getResults

File: plugins/org.eclipse.osee.framework.core/src/org/eclipse/osee/framework/core/data/OrcsTokenServiceImpl.java (hunk 1 above)
Comment (audrey-denk) [reply plugins/org.eclipse.osee.framework.core/src/org/eclipse/osee/framework/core/data/OrcsTokenServiceImpl.java:66]: scratch that we only need 1, cause we are using Ids

**Review by lucianovaglienti — APPROVED** [review]

File: plugins/org.eclipse.osee.orcs.rest/src/org/eclipse/osee/orcs/rest/internal/RelationEndpointImpl.java [hunk 5]
```diff
@@ -91,6 +92,27 @@ public List<RelationTypeToken> convertRelations(ArtifactId artToken, RelationTyp
...
+         oldRelationType).getResults().getList()) {
+         if (art.getRelatedCount(sideB) > 0) {
+            for (ArtifactReadable artB : art.getRelated(sideB).getList()) {
+               tx.relate(art, newRelationType, artB);
+            }
+         }
+      }
+      tx.commit();
+
+      return null;
```
Comment (lucianovaglienti) [inline plugins/org.eclipse.osee.orcs.rest/src/org/eclipse/osee/orcs/rest/internal/RelationEndpointImpl.java:113]: Q: Can we make this return the results of tx.commit()?

Comment (audrey-denk) [reply plugins/org.eclipse.osee.orcs.rest/src/org/eclipse/osee/orcs/rest/internal/RelationEndpointImpl.java:113]: of course, I should have caught that.

**Pull request merged**